test/e2e/auth/projected_clustertrustbundle.go

```diff
@@ -46,84 +49,287 @@ var _ = SIGDescribe(feature.ClusterTrustBundle, feature.ClusterTrustBundleProjec
 	f := framework.NewDefaultFramework("projected-clustertrustbundle")
 	f.NamespacePodSecurityLevel = admissionapi.LevelBaseline
 
-	goodCert1 := mustMakeCertificate(&x509.Certificate{
-		SerialNumber: big.NewInt(0),
-		Subject: pkix.Name{
-			CommonName: "root1",
-		},
-		IsCA:                  true,
-		BasicConstraintsValid: true,
-	})
+	var certPEMs []string
+	for i := range 10 {
+		certPEMs = append(certPEMs, mustMakeCAPEM(fmt.Sprintf("root%d", i)))
+	}
+
+	// We must copy the original fields. Slicing is just applied pointer arithmetics
+	// and if we attempted an append to such a slice later, we'd be overriding the original
+	// underlying array.
+	signerOnePEMS := make([]string, 8)
+	copy(signerOnePEMS[0:4], certPEMs[0:4])
+	copy(signerOnePEMS[4:], certPEMs[6:])
```

Comment:
I'll think about how to make the init logic easier on the reader

Reply:
I made `mustInitCTBs()` return a map of PEM sets so that you can operate on 2 dimensions (signer name and labels) on it by using set intersection. See if it makes sense to you.

Reply:
Slack [thread](https://kubernetes.slack.com/archives/C05MW3382US/p1724860371610649).